include/sqlpp11/data_types/unsigned_integral/expression_operators.h

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) 2013-2015, Roland Bock
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without modification,
+ * are permitted provided that the following conditions are met:
+ *
+ *   Redistributions of source code must retain the above copyright notice, this
+ *   list of conditions and the following disclaimer.
+ *
+ *   Redistributions in binary form must reproduce the above copyright notice, this
+ *   list of conditions and the following disclaimer in the documentation and/or
+ *   other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef SQLPP_UNSIGNED_INTEGRAL_EXPRESSION_OPERATORS_H
+#define SQLPP_UNSIGNED_INTEGRAL_EXPRESSION_OPERATORS_H
+
+#include <sqlpp11/expression_return_types.h>
+#include <sqlpp11/operand_check.h>
+#include <sqlpp11/expression_operators.h>
+#include <sqlpp11/basic_expression_operators.h>
+#include <sqlpp11/value_type.h>
+#include <sqlpp11/type_traits.h>
+#include <sqlpp11/data_types/unsigned_integral/data_type.h>
+
+namespace sqlpp
+{
+  template <typename Expression>
+  struct expression_operators<Expression, unsigned_integral> : public basic_expression_operators<Expression>
+  {
+  };
+
+  template <typename L, typename R>
+  struct return_type_plus<L, R, binary_operand_check_t<L, is_unsigned_integral_t, R, is_numeric_t>>
+  {
+    using check = consistent_t;
+    using type = plus_t<wrap_operand_t<L>, value_type_of<wrap_operand_t<R>>, wrap_operand_t<R>>;
+  };
+
+  template <typename L, typename R>
+  struct return_type_minus<L, R, binary_operand_check_t<L, is_unsigned_integral_t, R, is_numeric_t>>
+  {
+    using check = consistent_t;
+    using type = minus_t<wrap_operand_t<L>, value_type_of<wrap_operand_t<R>>, wrap_operand_t<R>>;
```

Comment:
-3 + 1 = -2
-1 * 5 = -5
-2 / 1 = -2
sqlite3 behaves this way, idk about what ANSI SQL says (or it unsigned is part of ANSI SQL). Also, in MySQL (specifically MariaDB, but MySQL is probably the same) if you subtract an unsigned column from an unsigned column it throws an error if the result would be negative and the . This is different than sqlite3 which just returns a signed version.

I did a commit to fix integrals and unsigned integrals so that any operation between a signed and unsigned will yield a signed (except modulus which I left to same type), and added tests for unsigned to make sure that any type and any op with an unsigned yields the other type and any time you subtract or unary minus an unsigned then it is signed. I added a type trait for is_numeric_not_unsigned_t and specialized some of the expressions for integral when working with unsigned to return signed.